internal/backends/linuxkms/renderer/sw.rs

```diff
@@ -137,4 +147,21 @@ impl crate::fullscreenwindowadapter::FullscreenRenderer for SoftwareRendererAdap
     ) -> Result<(), PlatformError> {
         self.display.drm_output.register_page_flip_handler(event_loop_handle)
     }
+
+    fn grab_window(&self) -> Result<SharedImageBuffer, PlatformError> {
+        let width = self.size.width;
+        let height = self.size.height;
+
+        // We could use Surface::fetch() here, but it's only implemented on X11 and Web.
+
+        let mut target_buffer =
+            SharedPixelBuffer::<i_slint_core::graphics::Rgb8Pixel>::new(width, height);
+
+        self.force_next_frame_new_buffer.set(true);
+        self.renderer.set_repaint_buffer_type(RepaintBufferType::NewBuffer);
+
+        let _ = self.renderer.render(target_buffer.make_mut_slice(), width as usize);
+
+        Ok(SharedImageBuffer::RGB8(target_buffer))
+    }
```

Comment:
This, along with that cell, is duplicated in the linuxkms adapter. I could also have a `fn screenshot()` function in the public software renderer API - like with femtovg.